### ISSUE TITLE
Ensure all components specify box-sizing to avoid bugs due to external styles

### ISSUE THREAD

**Issue by kfranqueiro** [material-components/material-components-web#1455]
We could probably manually test this by throwing in a `* { box-sizing: ... }` rule (set to content-box and border-box) in `demos/common.scss` to see if anything obviously breaks.

It looks like several components already account for box-sizing but I'll leave everything unchecked here until we actually test.

* [ ] mdc-button
* [ ] mdc-card
* [ ] mdc-checkbox
* [ ] mdc-dialog
* [ ] mdc-drawer
* [ ] mdc-fab
* [ ] mdc-form-field
* [ ] mdc-grid-list
* [ ] mdc-icon-toggle
* [ ] mdc-layout-grid
* [ ] mdc-linear-progress
* [ ] mdc-list
* [ ] mdc-menu
* [ ] mdc-radio
* [ ] mdc-ripple
* [ ] mdc-select
* [ ] mdc-slider
* [ ] mdc-switch
* [ ] mdc-tabs
* [ ] mdc-textfield
* [ ] mdc-toolbar

**Comment by RobertSmits** [comment]
I have checked all demo pages for errors relating to box-sizing.
Now these are fixed but there are still two "errors".
- mdc-list: 
The demo list is a bit smaller due to css specific to the demo page.
- mdc-dialog:
I have not done any work on this but when setting ``* { box-sizing: border-box }`` the body of the scrollable dialog gets a little smaller. Reason for not jet fixing this is #1442 

What should I do with these two?

**Comment by kfranqueiro** [comment]
Thanks for looking into this.

RE mdc-list, I wouldn't be particularly worried about the hero styles in the demo page, but switching to border-box does affect how the max-width on all lists is applied. However, I'd have to check whether the max-width we set is arbitrary, since I don't see mention of width in the guidelines.

I'm curious if there's a reason you chose border-box for mdc-list in your branch, whereas you chose content-box for the others? Wouldn't using content-box for mdc-list preserve its current behavior? (Using border-box necessitates the extra `calc` usage as well which could be a little harder to follow than the current alternative.)

RE dialog, I think applying border-box will end up affecting how the height of the scrolling container behaves, since it also has padding. Granted, that height is already arbitrary so this might not matter, but I'd have the same question as above RE choosing border-box vs. preserving the current content-box behavior?

**Comment by kfranqueiro** [comment]
Whoops, apologies, I just realized what you meant about the list changes really only affecting demo-specific styles. I wouldn't be too worried about affecting demo-only styles, though my concern still stands RE making the width style more complicated for list items with ripple.

**Comment by RobertSmits** [comment]
Most components (almost all) already use border-box, so where possible I wanted to continue this.
For this reason I edited the mdc-list to border-box.

The mdc-grid-list is quite complicated in current layout and I couldn't get it to work without adding a constant value in js, which needs to be changed when the scss counterpart is changed, or by getting css values in js which can cause issues with async css like that is the case with webpack.

So basically I didn't want to complicate these things too much and chose content-box for them.

**Comment by sidot3291** [comment]
This declaration is breaking mdc-select:
```
*,
*::before,
*::after {
  box-sizing: border-box;
}
```

Specifically, arrow positioning is too low:

Without override: <img width="77" alt="screen shot 2018-04-04 at 4 18 48 pm" src="https://user-images.githubusercontent.com/614945/38333215-f4ed9bda-3825-11e8-978f-60718767b8c9.png">

With override: <img width="80" alt="screen shot 2018-04-04 at 4 17 27 pm" src="https://user-images.githubusercontent.com/614945/38333225-fbbb09de-3825-11e8-93c3-9ffa6a3d4bda.png">

**Comment by sir-boformer** [comment]
![grafik](https://user-images.githubusercontent.com/22688477/40247641-5e41ab8e-5acd-11e8-8a45-d2235ad14eab.png)

The new outlined text field is also broken when box-sizing is globally set to `border-box`

**Comment by JetFault** [comment]
All TextFields are broken with `border-box`

![image](https://user-images.githubusercontent.com/487245/40507532-b299bd92-5f65-11e8-8006-a94db6257aaa.png)

**Comment by abhiomkar** [comment]
Closing this as obsolete.
